distribution/packages/src/deb/init.d/elasticsearch

```diff
@@ -81,16 +81,16 @@ if [ ! -x "$DAEMON" ]; then
 fi
 
 checkJava() {
-	if [ -x "$JAVA_HOME/bin/java" ]; then
-		JAVA="$JAVA_HOME/bin/java"
-	else
-		JAVA=`which java`
-	fi
-
-	if [ ! -x "$JAVA" ]; then
-		echo "Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME"
-		exit 1
-	fi
+  if [ ! -z "${JAVA_HOME}" ]; then
```

Comment:
We don't seem to actually use JAVA here anywhere. Other than exporting JAVA_HOME, could we remove this logic and let bin/elasticsearch handle it?

Reply:
I'll explore the consequences of this in a follow-up, I want to get this fix in for 7.3.1.